### PR TITLE
mixer: sync store before runtime watch

### PR DESCRIPTION
**Pull request by kyessenov**
It appears runtime starts watching in the constructor prior to full store sync.
That causes dangling reference errors log noise before the full store sync.
Make init explicit, and move it with sync before runtime watch start.

/assign @mandarjog 

Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

**Comment by kyessenov** [comment]
Sample noise:
```
2018-08-17T21:56:39.610935Z	warn	Failed to discover kinds: [dogstatsd cloudwatch]
2018-08-17T21:56:39.611079Z	info	Built new config.Snapshot: id='1'
2018-08-17T21:56:39.611164Z	info	Cleaning up handler table, with config ID:0
2018-08-17T21:56:39.611173Z	info	Awaiting for config store sync...
2018-08-17T21:56:40.641390Z	info	Publishing 8 events
2018-08-17T21:56:40.641622Z	error	failed to evaluate expression for field 'Value'; unknown attribute response.size
2018-08-17T21:56:40.641651Z	error	failed to evaluate expression for field 'Value'; unknown attribute connection.sent.bytes
2018-08-17T21:56:40.641673Z	error	failed to evaluate expression for field 'Value'; unknown attribute connection.received.bytes
2018-08-17T21:56:40.641705Z	error	failed to evaluate expression for field 'Dimensions[destination_service_name]'; unknown attribute destination.service.name
2018-08-17T21:56:40.641727Z	error	failed to evaluate expression for field 'Value'; unknown attribute response.duration
2018-08-17T21:56:40.641741Z	error	failed to evaluate expression for field 'Value'; unknown attribute request.size
2018-08-17T21:56:40.641753Z	info	Built new config.Snapshot: id='2'
2018-08-17T21:56:40.641810Z	info	Cleaning up handler table, with config ID:1
2018-08-17T21:56:42.224685Z	info	Publishing 9 events
2018-08-17T21:56:42.232250Z	error	Instance not found: instance='accesslog.logentry'
2018-08-17T21:56:42.232327Z	error	No valid instances found
2018-08-17T21:56:42.232332Z	error	No valid actions found in rule
2018-08-17T21:56:42.232376Z	error	Handler not found: handler='handler.prometheus'
2018-08-17T21:56:42.232386Z	error	No valid actions found in rule
2018-08-17T21:56:42.232425Z	error	Instance not found: instance='tcpaccesslog.logentry'
2018-08-17T21:56:42.232440Z	error	No valid instances found
2018-08-17T21:56:42.232445Z	error	No valid actions found in rule
2018-08-17T21:56:42.232514Z	error	Handler not found: handler='handler.prometheus'
2018-08-17T21:56:42.232533Z	error	No valid actions found in rule
2018-08-17T21:56:42.232540Z	info	Built new config.Snapshot: id='3'
2018-08-17T21:56:42.232718Z	info	adapters	getting kubeconfig from: ""	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T21:56:42.232737Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2018-08-17T21:56:42.236659Z	info	adapters	Waiting for kubernetes cache sync...	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T21:56:42.341875Z	info	adapters	Cache sync successful.	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T21:56:42.342870Z	info	Cleaning up handler table, with config ID:2
2018-08-17T21:56:44.625400Z	info	Publishing 3 events
2018-08-17T21:56:44.627841Z	info	Built new config.Snapshot: id='4'
2018-08-17T21:56:44.628674Z	info	adapters	getting kubeconfig from: ""	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T21:56:44.628747Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2018-08-17T21:56:44.628839Z	info	adapters	serving prometheus metrics on 42422	{"adapter": "handler.prometheus.istio-system"}
2018-08-17T21:56:44.630065Z	info	adapters	Waiting for kubernetes cache sync...	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T21:56:44.730240Z	info	adapters	Cache sync successful.	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T21:56:44.741150Z	info	Cleaning up handler table, with config ID:3
2018-08-17T21:56:44.741211Z	error	adapters	adapter did not close all the scheduled daemons	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T21:56:45.029449Z	info	Starting monitor server...
```

**Comment by mandarjog** [comment]
How does the log look after the fix?

**Comment by kyessenov** [comment]
```
2018-08-17T23:38:57.161109Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2018-08-17T23:38:57.207784Z	warn	Failed to discover kinds: [dogstatsd cloudwatch]
2018-08-17T23:38:57.207832Z	info	Awaiting for config store sync...
2018-08-17T23:39:02.419271Z	info	Starting runtime config watch...
2018-08-17T23:39:02.419424Z	info	Built new config.Snapshot: id='0'
2018-08-17T23:39:02.419492Z	info	Cleaning up handler table, with config ID:-1
2018-08-17T23:39:02.476660Z	info	Built new config.Snapshot: id='1'
2018-08-17T23:39:02.476772Z	info	adapters	getting kubeconfig from: ""	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T23:39:02.476785Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2018-08-17T23:39:02.477639Z	info	adapters	Waiting for kubernetes cache sync...	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T23:39:02.678405Z	info	adapters	Cache sync successful.	{"adapter": "handler.kubernetesenv.istio-system"}
2018-08-17T23:39:02.680132Z	info	adapters	serving prometheus metrics on 42422	{"adapter": "handler.prometheus.istio-system"}
2018-08-17T23:39:02.682133Z	info	Cleaning up handler table, with config ID:0
2018-08-17T23:39:03.150440Z	info	Starting monitor server...
```

File: mixer/pkg/server/server.go [hunk 1]
```diff
@@ -199,13 +198,9 @@ func newServer(a *Args, p *patchTable) (*Server, error) {
...
-	// this method initializes the store
-	if err = p.runtimeListen(rt); err != nil {
-		_ = s.Close()
-		return nil, fmt.Errorf("unable to listen: %v", err)
+	kinds := runtimeconfig.KindMap(adapterMap, templateMap)
+	if err := st.Init(kinds); err != nil {
+		return nil, fmt.Errorf("unable to initialize config store: %v", err)
 	}
 
 	// block wait for the config store to sync
```
Comment (mandarjog) [inline mixer/pkg/server/server.go]: nit: `Awaiting config store sync...`

**Comment by kyessenov** [comment]
Please re-LGTM @mandarjog . I rebased to pick up the test fix.

**Pull request merged**